flowauth/frontend/cypress/integration/add_new_token.js

```diff
@@ -54,6 +54,165 @@ describe("Token generation", function() {
+    cy.get("#api-exp").click();
+    //unchecking first sub-level checkbox
+    cy.get("#permission")
+      .first()
+      .click({
+        force: true
+      });
+    //unchecking first sub-level checkbox
+    cy.get("#permission")
+      .first()
+      .click({
+        force: true
+      });
+    //checking top-level permission checkbox
+    cy.get("#permissions").should("be.checked");
+  });
+  it("Top-level API permisions checkbox unchecked", function() {
+    cy.get("#new").click();
+    //unchecking permission top level checkbox
+    cy.get("#permissions").click();
+    cy.get("#api-exp").click();
+    //checking first sub-level checkbox
+    cy.get("#permission")
+      .first()
+      .click({
+        force: true
+      });
+    //checking first sub-level checkbox
+    cy.get("#permission")
+      .first()
+      .click({
+        force: true
+      });
+
+    cy.get("#permissions").should("not.be.checked");
+  });
+  it("Top-level API permisions checkbox intermidiate", function() {
+    cy.get("#new").click();
+    cy.get("#api-exp").click();
+    //checking first sub-level permission checkbox
+    cy.get("#permission")
+      .first()
+      .click({
+        force: true
+      });
+    cy.get("#permissions").should("have.attr", "data-indeterminate", "true");
+  });
+  it("Aggrigation unit sub-level checkboxs checked", function() {
+    cy.get("#new").click();
+    //uncheck aggrigation unit top-level checkbox
+    cy.get("#units").click();
+    //check aggrigation unit top-level checkbox
+    cy.get("#units").click();
+    cy.get("#unit-exp").click();
+    cy.get("#unit")
+      .first()
+      .should("be.checked");
+  });
+  it("Aggrigation unit sub-level checkboxs unchecked", function() {
+    cy.get("#new").click();
+    //uncheck aggrigation unit top-level checkbox
+    cy.get("#units").click();
+    cy.get("#unit-exp").click();
+    cy.get("#unit")
+      .first()
+      .should("not.be.checked");
+  });
+  it("Top-level Aggrigation unit checkbox checked", function() {
+    cy.get("#new").click();
+    cy.get("#unit-exp").click();
+    //uncheck first sub-level checkbox
+    cy.get("#unit")
+      .first()
+      .click({
+        force: true
+      });
+    //check first sub-level checkbox
+    cy.get("#unit")
+      .first()
+      .click({
+        force: true
+      });
+    cy.get("#units").should("be.checked");
+  });
+  it("Top-level Aggrigation unit checkbox unchecked", function() {
+    cy.get("#new").click();
+    //uncheck aggrigation unit top-level checkbox
+    cy.get("#units").click();
+    cy.get("#unit-exp").click();
+    //check first sub-level checkbox
+    cy.get("#unit")
+      .first()
+      .click({
+        force: true
+      });
+    //uncheck first sub-level checkbox
+    cy.get("#unit")
+      .first()
+      .click({
+        force: true
+      });
+    cy.get("#units").should("not.be.checked");
+  });
+  it("Top-level Aggrigation unit checkbox intermidiate", function() {
```

Comment:
```suggestion
  it("Top-level Aggregation unit checkbox indeterminate", function() {
```

Can these be fixed everywhere? I know it is a bit anal, but I find the typos really distracting! See also: checkboxs when it should be checkboxes.